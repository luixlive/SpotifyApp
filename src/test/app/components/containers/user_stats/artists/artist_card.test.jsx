import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';

import {
  artistCard as componentProps,
} from './../../../../../test_utils/components_props';
import initialState from './../../../../../test_utils/initial_state';
import ArtistCard, {
  formatGenres,
  PureArtistCard,
/* eslint-disable-next-line max-len */
} from './../../../../../../app/components/containers/user_stats/artists/artist_card';

describe('App Components - Artist Card', () => {
  describe('Helpers', () => {
    it('formats genres with a break between them', () => {
      const genres = ['genre 1', 'genre 2'];
      const genresSeparatedByBreaks = formatGenres(genres);

      const wrapper = shallow(<div>{genresSeparatedByBreaks}</div>);
      expect(wrapper.text().includes('genre 1')).toBeFalsy();
      expect(wrapper.text().includes('Genre 1')).toBeTruthy();
      expect(wrapper.text().includes('genre 2')).toBeFalsy();
      expect(wrapper.text().includes('Genre 2')).toBeTruthy();
      expect(wrapper.find('br').length).toBe(2);
    });
  });

  describe('Snapshots', () => {
    let props;
    beforeEach(() => {
      props = _.cloneDeep(componentProps);
    });

    it('renders no mobile', () => {
      const rendered = renderer.create(<PureArtistCard {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders mobile', () => {
      props.deviceMobile = true;
      const rendered = renderer.create(<PureArtistCard {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Provider', () => {
    const mockStore = configureStore();
    let store;
    let wrapper;
    beforeAll(() => {
      store = mockStore(initialState);
      wrapper = mount((
        <Provider store={store}>
          <ArtistCard place={0} />
        </Provider>
      ));
    });

    it('renders', () => {
      expect(wrapper.find(PureArtistCard).length).toEqual(1);
    });

    it('matches initial state', () => {
      expect(Object.keys(wrapper.find(PureArtistCard).props()))
        .toEqual(Object.keys(componentProps));
    });
  });
});
