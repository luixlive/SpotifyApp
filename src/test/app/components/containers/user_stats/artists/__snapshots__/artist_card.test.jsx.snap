// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - Artist Card Snapshots renders mobile 1`] = `
<div>
  <h3
    className="ui left floated header"
  >
    0
  </h3>
  <div
    className="ui fluid card artist"
    href={undefined}
    onClick={[Function]}
  >
    <a
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        className="ui image"
        src=""
      />
    </a>
    <div
      className="content"
    >
      <div
        className="header"
      >
        <h4
          className="ui header"
        >
          <a
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            
          </a>
        </h4>
      </div>
      <div
        className="meta"
      >
        <h5
          className="sub header"
        />
      </div>
    </div>
    <div
      className="extra content"
    >
      Followers: 
      0
    </div>
  </div>
</div>
`;

exports[`App Components - Artist Card Snapshots renders no mobile 1`] = `
<div>
  <h2
    className="ui left floated header"
  >
    0
  </h2>
  <div
    className="ui fluid card artist"
    href={undefined}
    onClick={[Function]}
  >
    <a
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        className="ui image"
        src=""
      />
    </a>
    <div
      className="content"
    >
      <div
        className="header"
      >
        <h3
          className="ui header"
        >
          <a
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            
          </a>
        </h3>
      </div>
      <div
        className="meta"
      >
        <h4
          className="sub header"
        />
      </div>
    </div>
    <div
      className="extra content"
    >
      Followers: 
      0
    </div>
  </div>
</div>
`;
